h2o-core/src/main/java/hex/RegexTokenizer.java

```diff
@@ -0,0 +1,98 @@
+package hex;
+
+import water.MRTask;
+import water.fvec.Chunk;
+import water.fvec.Frame;
+import water.fvec.NewChunk;
+import water.fvec.Vec;
+import water.parser.BufferedString;
+
+/**
+ * RegexTokenizer splits rows of a given Frame into delimited sequences of tokens using a regular expression.
+ * The output structure is suitable for use in the word2vec algorithm.
+ *
+ * <p>
+ * Example usage:
+ * <pre>{@code
+ * final RegexTokenizer tokenizer = new RegexTokenizer.Builder()
+ *   .setRegex("[,;]")
+ *   .setMinLength(2)
+ *   .setToLowercase(true)
+ *   .create();
+ * final Frame tokens = tokenizer.transform(inputFrame);
+ * }
+ * </pre>
+ */
+public class RegexTokenizer extends MRTask<RegexTokenizer> {
```

Comment:
WDYT about exposing a property that would enable to specify what columns will be used for tokenization? Couldn't  a frame contain also other features that are not related to word2vec?

Reply:
Instead of doing that user can just subset the frame's columns. IMHO that is a better approach from maintainability point of view.

Example:

    final Frame tokens = tokenizer.transform(inputFrame.subframe(new String[]{"a", "b"}));

Reply:
Ok and other features can be appended to the output of word2vec by some `join` operation, I assume?

Reply:
IMHO that is not necessary (or possible), you only use the tokenizer for word2vec training (as a direct input to the model builder). If you have multiple algos, then they will have their own inputs. Perhaps this is the reason why this tokenizer is not good for SW integration. However, imho it makes sense in H2O's context.

If you want to use word2vec's transform then you can just run Word2VecModel#transform() and column bind the rest of the features if they are supposed to be used as an input of a dependent model.

Reply:
Thanks for clarification!